### PR TITLE
Store index ID to filter out invalid handoff event

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#19792

### CODE REVIEW

**Comment by yah01** [comment]
subsequent PR of #19516

File: internal/util/typeutil/map.go [hunk 1]
```diff
@@ -2,6 +2,16 @@ package typeutil
...
 
+// MapEqual returns true if the two map contain the same keys and values
+func MapEqual[K comparable, V any](left, right map[K]V) bool {
+	for k, v := range left {
+		if v2, ok := right[k]; !ok || v != v2 {
+			return false
+		}
+	}
+	return true
+}
```
Comment (congqixia) [inline internal/util/typeutil/map.go:13]: Only has `right contains left` logic.
Need unit tests for this important logic utility function.

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**